base/abstractarray.jl

```diff
@@ -2403,7 +2403,14 @@ function hash(A::AbstractArray, h::UInt)
     # Instead hash the tuple of firsts and lasts along each dimension
     h = hash(map(first, axes(A)), h)
     h = hash(map(last, axes(A)), h)
-    isempty(A) && return h
+
+    # For short arrays, it's not worth doing anything complicated 
+    if length(A) < 10000
```

Comment:
Any rough tests on this threshold?  Seems like we'd want it to jive with the 4096 below.

Reply:
For a `10000` length `Vector{Float64}`, I'm seeing a 7% speedup (more for matrices of the same size, not sure why).  The reason this can still be beneficial is that the logic is enough simpler that it can still be faster while doing 50% more hashes. That said, the tradeoff will be less good for `Array`s of more complex objects. Given that, I could see arguments both ways. What are your thoughts?